aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java

```diff
@@ -91,8 +116,8 @@ public void blogOwnerSavedIntoDataStoreIsThenQueriableInRemoteAppSyncApi() throw
             .build();
         dataStore.save(localCharley);
 
-        // Wait a bit. TODO: this is lame; how to tell deterministically when sync engine has sync'd?
-        Sleep.milliseconds(NETWORK_OP_TIMEOUT_MS + DATA_STORE_OP_TIMEOUT_MS);
+        // Wait for a Hub event telling us that our Charley model got published to the cloud.
+        outboundModelEventAccumulator.takeOne();
```

Comment:
What's a Charley model?

Reply:
Oh I see, `.name("Charley Crockett")`
I was thinking this was a new data structure 😂

Reply:
Haha we should invent it!!